pkg/driver/node/node.go

```diff
@@ -115,6 +118,18 @@ func (ns *S3NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePubl
 	}
 
 	args := mountpoint.ParseArgs(mountpointArgs)
+
+	if capMount := volCap.GetMount(); capMount != nil && util.UsePodMounter() {
+		if volumeMountGroup := capMount.GetVolumeMountGroup(); capMount.GetVolumeMountGroup() != "" {
+			// We need to add the following flags to support fsGroup
+			// If these flags were already set by customer in PV mountOptions then we won't override them
+			args.SetIfAbsent(mountpoint.ArgGid, volumeMountGroup)
+			args.SetIfAbsent(mountpoint.ArgAllowOther, "")
+			args.SetIfAbsent(mountpoint.ArgDirMode, "770")
+			args.SetIfAbsent(mountpoint.ArgFileMode, "660")
```

Comment:
This block can update some arguments partially, should we prevent that? Maybe we should set these only if `gid` is not passed by the customer?

Reply:
Also, you can use `mountpoint.ArgNoValue` instead of empty string in `args.SetIfAbsent(mountpoint.ArgAllowOther, "")`, so it'd become `args.SetIfAbsent(mountpoint.ArgAllowOther, mountpoint.ArgNoValue)`